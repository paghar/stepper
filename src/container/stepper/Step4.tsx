import React, {useEffect} from "react";
import {updateFromStep} from "../../redux/user/userActions";
import {useSelector, useDispatch} from "react-redux";
import {IUser} from "../../data/interface";

const Step4 = () => {

    const usersData = useSelector(state => state as any)?.userState;
    const userInfo = usersData.users as IUser;  
    const dispatch = useDispatch(); 
    
    const itemText = (text:string) =>{
        return(
            <label className="block text-gray-700 text-sm font-bold h-8">{text}</label> 
        );
    };

    useEffect(()=>{           
        dispatch(updateFromStep(4));
    },[]);

   
    return (
        <div className="bg-white shadow-md rounded p-12 m-10 break-words lg:text-base md:text-xs sm:text-xs text-xs"> 

            {itemText(`Step 1`)}              
            <div className="ml-6">                
                {itemText(`Name:${userInfo?.name}`)}
                {itemText(`Email:${userInfo?.email}`)}
                {itemText(`Phone-number:${userInfo?.phone}`)}
                {itemText(`Gender:${userInfo?.gender}`)}               
            </div> 

            {itemText(`Step 2`)}           
            <div className="ml-6">
                {itemText(`Date of birth:${userInfo?.brithDay}`)}
                {itemText(`Address:${userInfo?.address}`)}
                {itemText(`Spoken Languages:${userInfo?.spokenLanguage}`)}
                {itemText(`Height:${userInfo?.height}`)}  
                {itemText(`Weight:${userInfo?.weight}`)}                 
            </div>                   
                
            {itemText(`Step 3`)}
            <div className="ml-6">
                {itemText(`Accept condition:${userInfo?.condation}`)}  
            </div>          
                
        </div>
    );
};

export default Step4;
