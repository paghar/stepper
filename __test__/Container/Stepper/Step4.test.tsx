import renderer from "react-test-renderer";
import Step4 from "../../../src/container/stepper/Step4";
import {Provider} from "react-redux";

import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore();

jest.mock("next/router", () => ({
    useRouter: jest.fn()
}));

describe(">>>> Stepper", () => {
    test("++++ Snapshot of Stepper4 form", () => {       
        const component = renderer.create(
            <Provider store={store}>
                <Step4/>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

export {};

