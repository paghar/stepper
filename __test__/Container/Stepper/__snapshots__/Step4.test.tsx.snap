// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>> Stepper ++++ Snapshot of Stepper4 form 1`] = `
<div
  className="bg-white shadow-md rounded p-12 m-10 break-words lg:text-base md:text-xs sm:text-xs text-xs"
>
  <label
    className="block text-gray-700 text-sm font-bold h-8"
  >
    Step 1
  </label>
  <div
    className="ml-6"
  >
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Name:undefined
    </label>
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Email:undefined
    </label>
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Phone-number:undefined
    </label>
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Gender:undefined
    </label>
  </div>
  <label
    className="block text-gray-700 text-sm font-bold h-8"
  >
    Step 2
  </label>
  <div
    className="ml-6"
  >
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Date of birth:undefined
    </label>
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Address:undefined
    </label>
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Spoken Languages:undefined
    </label>
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Height:undefined
    </label>
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Weight:undefined
    </label>
  </div>
  <label
    className="block text-gray-700 text-sm font-bold h-8"
  >
    Step 3
  </label>
  <div
    className="ml-6"
  >
    <label
      className="block text-gray-700 text-sm font-bold h-8"
    >
      Accept condition:undefined
    </label>
  </div>
</div>
`;
